services/ext/api.go

```diff
@@ -291,8 +291,8 @@ func (api *PublicAPI) Chats(parent context.Context) []*protocol.Chat {
 	return api.service.messenger.Chats()
 }
 
-func (api *PublicAPI) ChatsPreview(parent context.Context) []*protocol.ChatPreview {
-	return api.service.messenger.ChatsPreview()
+func (api *PublicAPI) ChatsPreview(parent context.Context, onlyCommunityChats *bool) []*protocol.ChatPreview {
```

Comment:
I had the impression the standard was to use structs for the request args in endpoints because then we can always add keys (and thus behavior) without breaking clients. Is there an agreement on this convention? cc @igor-sirotin

Reply:
@ilmotta this is true, but only for `mobile/status.go`. 

APIs in `services/**/api.go` are called through JSON-RPC server, so the behaviour in this case is a bit different, as we pass arguments as a JSON array. 

# TLDR;

Yes, for compatibility it is better to make `struct` arguments.

# Investigated details

I checked the behaviour for redundant/missing arguments.

1. **Redundant argument** - returns proper error

    Request: 
    ```json
    {
        "jsonrpc": "2.0",
        "method": "settings_getSettings",
        "params": ["redundant_arg"]
    }
    ```
    Response:
    ```json
    {
        "jsonrpc": "2.0",
        "id": 0,
        "error": {
            "code": -32602,
            "message": "too many arguments, want at most 0"
        }
    }
    ```

2. **Missing pointer argument** - uses `nil` value for missing argument

    Request:
    ```json
    {
        "jsonrpc": "2.0",
        "method": "wakuext_acceptContactRequest",
        "params": []
    }
    ```
    
    Response:
    ```json
    {
        "jsonrpc": "2.0",
        "id": 0,
        "error": {
            "code": -32603,
            "message": "method handler crashed"
        }
    }
    ```
    
    There was panic, because `request` argument is `nil` in
    https://github.com/status-im/status-go/blob/17dbc379641f9c54772df09ab10a4979ad9e968d/services/ext/api.go#L857-L859

3. **Missing non-pointer argument** - returns proper error
    
    `SetUserStatus` expects 2 arguments:
    https://github.com/status-im/status-go/blob/17dbc379641f9c54772df09ab10a4979ad9e968d/services/ext/api.go#L802-L804
    
    Request wihout `customText`:
    ```json
    {
        "jsonrpc": "2.0",
        "method": "wakuext_setUserStatus",
        "params": [
            1
        ]
    }
    ```
    
    Response:
    ```json
    {
        "jsonrpc": "2.0",
        "id": 0,
        "error": {
            "code": -32602,
            "message": "missing value for required argument 1"
        }
    }
    ```